atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -346,36 +371,37 @@ def set_group_manifest_info(self, extra):
             extra['image']['index'] = index
         # group_manifests returns None if didn't run, {} if group=False
         else:
-            for platform in self._worker_metadatas:
-                if platform == "x86_64":
-                    for instance in self._worker_metadatas[platform]['output']:
-                        if instance['type'] == 'docker-image':
-                            # koji_upload, running in the worker, doesn't have the full tags
-                            # so set them here
-                            instance['extra']['docker']['tags'] = tags
-                            instance['extra']['docker']['floating_tags'] = floating_tags
-                            instance['extra']['docker']['unique_tags'] = unique_tags
-                            repositories = []
-                            for pullspec in instance['extra']['docker']['repositories']:
-                                if '@' not in pullspec:
-                                    image = ImageName.parse(pullspec)
-                                    image.tag = version_release
-                                    pullspec = image.to_str()
-
-                                repositories.append(pullspec)
-
-                            instance['extra']['docker']['repositories'] = repositories
-                            self.log.debug("reset tags to so that docker is %s",
-                                           instance['extra']['docker'])
-                            # OSBS2 TBD: `get_worker_build_info` is imported from
-                            # build_orchestrate_build
-                            annotations = get_worker_build_info(self.workflow, platform).\
-                                build.get_annotations()
-
-                            digests = {}
-                            if 'digests' in annotations:
-                                digests = get_digests_map_from_annotations(annotations['digests'])
-                                instance['extra']['docker']['digests'] = digests
+            platform = "x86_64"
+            _, instance = next(
+                self._iter_work_metadata_outputs(platform, {"type": "docker-image"}),
+                (None, None),
+            )
+
+            if instance:
```

Comment:
Nice! This removes 3 levels of indentation and makes it clearer what we're actually doing :tada: :taco: